src/store/checkout-store.js

```diff
@@ -58,4 +65,31 @@ store.registerHandler('NEW_CHECKOUT', data => {
     }
 });
 
+store.registerHandler('NEW_LONGTERM_CHECKOUT', data => {
+    let longTermCheckout = {
+        studentId: data.studentId,
+        itemAddresses: data.itemAddresses,
+        longTermDueDate: data.longTermDueDate,
+        longtermProfessor: data.longtermProfessor
+    };
+
+    data.itemAddresses.forEach(itemAddress => {
+        if (ItemStore.getItemByAddress(itemAddress).status !== 'AVAILABLE') {
+            throw new Error('An item in the cart is not available for checkout.');
+        }
+    });
+
+    if (data.adminCode){
+        if(AuthStore.verifyAdmin(data.adminCode)) {
+            longTermCheckouts[data.actionId] = longTermCheckout;
+        } else {
+            throw new Error('Invalid Admin');
+        }
+    } else if (StudentStore.hasOverdueItem(data.studentId)) {
+        throw new Error('Student has overdue item');
+    } else {
+        longTermCheckouts[data.actionId] = longTermCheckout;
+    }
```

Comment:
Nit: This could be simplified.

```javascript
if (data.adminCode) {
    if (!AuthStore.verifyAdmin(data.adminCode)) {
        throw new Error('Invalid Admin');
    }
} else if (StudentStore.hasOverdueItem(data.studentId)) {
    throw new Error('Student has overdue item');
}
longTermCheckouts[data.actionId] = longTermCheckout;
```